packages/legend-application-data-cube/src/stores/CachedDataCubeSourceBuilder.ts

```diff
@@ -0,0 +1,165 @@
+/**
+ * Copyright (c) 2020-present, Goldman Sachs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { TDSBuilder } from '../../../legend-graph/src/graph-manager/action/execution/ExecutionResult.js';
+import {
+  V1_Binary,
+  V1_ClassInstance,
+  V1_ClassInstanceType,
+  V1_Column,
+  V1_Database,
+  V1_Date,
+  V1_Double,
+  V1_DuckDBDatasourceSpecification,
+  V1_EngineRuntime,
+  V1_IdentifiedConnection,
+  V1_Integer,
+  V1_RelationStoreAccessor,
+  V1_PackageableElementPointer,
+  V1_PackageableRuntime,
+  V1_PureModelContextData,
+  V1_RelationalDataType,
+  V1_RelationalDatabaseConnection,
+  V1_Schema,
+  V1_StoreConnections,
+  V1_Table,
+  V1_TestAuthenticationStrategy,
+  V1_VarChar,
+  V1_Lambda,
+} from '@finos/legend-graph';
+import type { LegendQueryDataCubeSource } from './model/LegendQueryDataCubeSource.js';
+import { CachedDataCubeSource } from '@finos/legend-data-cube';
+
+export function synthesizeCachedSource(
```

Comment:
I'm a big fan of not creating too many files to have to jump around in the codebase, I think we can really shorten the logic here into 1 single function and put that in `LegendDataCubeDataCubeEngine`. Not a must, but if you have time, let's do that